autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -1,23 +1,24 @@
 import * as React from 'react';
-import { RouteComponentProps } from '@reach/router';
 import { useSelector, useDispatch } from 'react-redux';
 import * as styles from './Schedule.css';
-import Meeting from '../../types/Meeting';
+import Meeting from '../../../types/Meeting';
 import MeetingCard from './MeetingCard/MeetingCard';
-import { RootState } from '../../redux/reducer';
-import { addAvailability, updateAvailability, mergeAvailability } from '../../redux/actions/availability';
-import { setSelectedAvailability, mergeThenSelectAvailability } from '../../redux/actions/selectedAvailability';
-import Availability, { AvailabilityType, AvailabilityArgs, roundUpAvailability } from '../../types/Availability';
+import { RootState } from '../../../redux/reducer';
+import { addAvailability, updateAvailability, mergeAvailability } from '../../../redux/actions/availability';
+import { setSelectedAvailability, mergeThenSelectAvailability } from '../../../redux/actions/selectedAvailability';
+import Availability, { AvailabilityType, AvailabilityArgs, roundUpAvailability } from '../../../types/Availability';
 import AvailabilityCard from './AvailabilityCard/AvailabilityCard';
 import HoveredTime from './HoveredTime/HoveredTime';
-import { FIRST_HOUR, LAST_HOUR, formatTime } from '../../timeUtil';
+import { FIRST_HOUR, LAST_HOUR, formatTime } from '../../../timeUtil';
 
-const Schedule: React.FC<RouteComponentProps> = () => {
+const Schedule: React.FC = () => {
   // these must be unique because of how they're used below
   const DAYS_OF_WEEK = ['M', 'T', 'W', 'R', 'F'];
 
   // "props" derived from Redux store
-  const schedule = useSelector<RootState, Meeting[]>((state) => state.meetings);
+  const schedule = useSelector<RootState, Meeting[]>(
+    (state) => state.schedules[state.selectedSchedule] || [],
+  );
```

Comment:
```suggestion
  const emptySchedule: Meeting[] = [];
  const schedule = useSelector<RootState, Meeting[]>(
    (state) => state.schedules[state.selectedSchedule] || emptySchedule,
  );
```
Since this returns a reference to a new list every time, it will rerender the schedule every time the state is changed if there isn't a schedule selected, which can be prevented if we create the reference beforehand

Reply:
Hmm, I think the local variable `emptySchedule` would also be re-created on every render, so I've used a static variable instead. That look good to you?

Also, once we have basic functionality, I think it would be interesting to write tests for optimization, to check if expensive calculations were being repeated on re-renders. Just a thought for now.

Reply:
From testing locally (by putting what I suggested and `console.log`ging something in the `useMemo` for meetingsForDays to check if it rerenders), it seems to still work fine if you put `const emptySchedule` inside of the functional component, but since it's pretty much the same and you already put it outside it's fine the way you changed it to